containers/glassfish/jersey-gf-ejb/src/main/java/org/glassfish/jersey/gf/ejb/internal/EjbComponentProvider.java

```diff
@@ -93,23 +94,43 @@
 
         final InitialContext ctx;
         final Class<T> clazz;
+        final String beanName;
         final EjbComponentProvider ejbProvider;
 
         @SuppressWarnings("unchecked")
         @Override
         public T get() {
             try {
-                return (T) lookup(ctx, clazz, clazz.getSimpleName(), ejbProvider);
+                return (T) lookup(ctx, clazz, beanName, ejbProvider);
             } catch (NamingException ex) {
                 Logger.getLogger(ApplicationHandler.class.getName()).log(Level.SEVERE, null, ex);
                 return null;
             }
         }
 
+        private static <T> String getBeanName(final Class<T> clazz) {
```

Comment:
Maybe you can explain why using the class name was not sufficient or wrong in this case?

Reply:
It's in tests that will be pushed into javaee7-samples. The reason is tha  name aatribute of Stateless/Stateful annotation is used to create JNDI name and the bean cannot be found only by the simple class name..

Reply:
I am certainly not a JNDI expert. I was thinking that one of the names of a bean is always the simple class name even if it gets named explicitly. Isn't that the case?

Reply:
It will be the case when there's no explicit name on the annotation, which is exactly what's being fixed here.